orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -470,4 +478,58 @@ class OperationsController {
       pipeline.origin = AuthenticatedRequest.spinnakerUserOrigin.orElse('unknown')
     }
   }
+
+  private void addStageAuthorizedRoles(def request, Map pipeline) {
+
+    def jsonObject = jsonSlurper.parseText(request)
```

Comment:
I'm kinda confused why you need to do any JSON slurping here to begin with. On L177, we do `def request = objectMapper.writeValueAsString(pipeline)`, and then on this line we're deserializing it again. It's wasteful, as you're passing the exact same original payload as `Map pipeline` into this method.

We can do without the json hoops entirely.

Reply:
Incorporated the change.